octane/evmengine/keeper/db.go

```diff
@@ -72,3 +72,40 @@ func (k *Keeper) updateExecutionHead(ctx context.Context, payload engine.Executa
 
 	return nil
 }
+
+// insertWithdrawal inserts a withdrawal request.
+func (k *Keeper) insertWithdrawal(ctx context.Context, withdrawalAddr common.Address, createdHeight uint64, amountGwei uint64) (uint64, error) {
+	id, err := k.withdrawalTable.InsertReturningId(ctx, &Withdrawal{
+		Address:       withdrawalAddr[:],
+		CreatedHeight: createdHeight,
+		AmountGwei:    amountGwei,
+	})
+	if err != nil {
+		return 0, errors.Wrap(err, "insert withdrawal")
+	}
+
+	withdrawals.Inc()
+
+	return id, nil
+}
+
+// getWithdrawals returns all stored withdrawals.
+func (k *Keeper) getWithdrawals(ctx context.Context) ([]*Withdrawal, error) {
```

Comment:
nit: `getAllWithdrawals`

Reply:
I guess this will only be used for testing, so can move this function to`db_internal_test.go`. That way the function is available only for testing,

Reply:
Will do. 

I think what we will need as well is a query for pending withdrawals for the given address to display all pending rewards...